### PR TITLE
fix(cosmlib): handle codec in val to eth address conversions

### PR DESCRIPTION
**Pull request by archbear**

### CODE REVIEW

File: cosmos/precompile/staking/staking.go [hunk 1]
```diff
@@ -246,10 +246,14 @@ func (c *Contract) GetDelegatorUnbondingDelegations(
 
 	unbondingDelegations := make([]generated.IStakingModuleUnbondingDelegation, 0)
 	for _, u := range res.GetUnbondingResponses() {
+		valAddress, err := sdk.ValAddressFromBech32(u.ValidatorAddress)
```
Comment (itsdevbear) [inline cosmos/precompile/staking/staking.go:249]: use AddressCodec

File: cosmos/precompile/staking/staking.go [hunk 2]
```diff
@@ -57,6 +63,7 @@ func NewPrecompileContract(sk *stakingkeeper.Keeper) *Contract {
 			generated.StakingModuleMetaData.ABI,
 			cosmlib.AccAddressToEthAddress(authtypes.NewModuleAddress(stakingtypes.ModuleName)),
 		),
+		ac:        sk,
```
Comment (itsdevbear) [inline cosmos/precompile/staking/staking.go:66]: storing 3 pointers to the same thing now kek, i guess its fine tho

**Review by itsdevbear — DISMISSED** [review]

File: cosmos/precompile/staking/staking.go (hunk 2 above)
Comment (archbear) [reply cosmos/precompile/staking/staking.go:66]: yea but no access cause interfaces

Comment (itsdevbear) [reply cosmos/precompile/staking/staking.go:66]: yeah no stress, it's kinda ugly no matter how u slice it

**Review by itsdevbear — DISMISSED** [review]

**Review by calbera — DISMISSED** [review]

**Review by calbera — APPROVED** [review]

**Review by ocnc — APPROVED** [review]
utACK

**Pull request merged**